libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -49,16 +49,9 @@ Declaration const* DeclarationContainer::conflictingDeclaration(
 		dynamic_cast<MagicVariableDeclaration const*>(&_declaration)
 	)
 	{
-		// check that all other declarations with the same name are functions or a public state variable or events.
-		// And then check that the signatures are different.
```

Comment:
This was a pure lie it seems... (at least inside this function - one could mention that this will happen later on, though)